zfsbootmenu/libexec/zfsbootmenu-init

```diff
@@ -78,6 +78,60 @@ elif [ ! -e /etc/hostid ]; then
   write_hostid "${default_hostid}"
 fi
 
+# Wait for devices to show up
+
+if [ -n "${zbm_waitfor_devices}" ]; then
+  IFS=',' read -r -a user_devices <<<"${zbm_waitfor_devices}"
+  while true; do
+    FOUND=0
+    EXPECTED=0
+    missing=()
+
+    for device in "${user_devices[@]}"; do
+      case "${device}" in
+        /dev/*)
+          ((EXPECTED++))
+          if [ -e "${device}" ] ; then
+            ((FOUND++))
+          else
+            missing+=( "$device" )
+          fi
+          ;;
+        *=*)
+          ((EXPECTED++))
+          path_prefix="/dev/disk/by-${device%=*}"
+          checkfor="${path_prefix,,}/${device##*=}"
+          if [ -e "${checkfor}" ] ; then
+            ((FOUND++))
+          else
+            missing+=( "$device" )
+          fi
+          ;;
```

Comment:
It occurs to me that devices can be specified just as dynamically with full `/dev/disk/by-*` paths, so this could be retained as a convenience that saves a few characters or discarded to reduce maintenance burden.

Reply:
I like the more compact form - it's easier to parse when reading through the ZBM KCL.  There's not too much of a maintenance burden with it, so I'll retain it unless you'd like it removed for less code.

Reply:
No, leaving it is fine and mimics fstab entries.